src/cloudant/views.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+"""
+_views_
+
+Utilities for handling design docs and the resulting views they create
+
+"""
+from .document import CloudantDocument
+
+
+class Code(str):
+    """
+    _Code_
+
+    string derived object that allows us to wrap/manipulate javascript blobs
+
+    """
+    def __init__(self, s):
+        super(Code, self).__init__(s)
+
+
+class View(dict):
+    """
+    Dictionary based object representing a view, exposing the map, reduce
+    functions as attributes and supporting query/data access via the view
+
+    """
+    def __init__(self, view_name, map_func=None, reduce_func=None):
+        super(View, self).__init__()
+        self.view_name = view_name
+        self[self.view_name] = {}
+        self[self.view_name]['map'] = map_func
+        self[self.view_name]['reduce'] = reduce_func
+        if self[self.view_name]['map'] and not isinstance(
```

Comment:
you could nix the bulk of this by providing a `_codify` function in this module which does the condition isinstance check. then this ctor boils down to:
```
self[self.view_name]['map'] = _codify(map_func)
...
```